***Currently working on the dataset from the Global Terrorism Database (GTD) maintained by the University of Maryland to gain insights into current state of terrorism in the US, and possibly making predictions of future unrewarded risks. I foresee this analysis valuable to the US Department of Homeland Security, other government establishments, businesses or individuals that care about safety of lives and properties.***

***Dataset can be found at: http://www.start.umd.edu/gtd/***

**Overseeing Mentor: Dr. Stylianos Kampakis***

In [1]:
import pandas as pd
import numpy as np

In [2]:
file= r'C:\Users\dejavu\Desktop\git_jupyter\springboard_mini_project/capstone_projects/globalterrorismdb_0617dist.xlsx'
df= pd.read_excel(file)

In [30]:
df.head()

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaT       58   
1  197000000002   1970       0     0        NaN         0        NaT      130   
2  197001000001   1970       1     0        NaN         0        NaT      160   
3  197001000002   1970       1     0        NaN         0        NaT       78   
4  197001000003   1970       1     0        NaN         0        NaT      101   

          country_txt  region   ...    addnotes scite1 scite2  scite3  \
0  Dominican Republic       2   ...         NaN    NaN    NaN     NaN   
1              Mexico       1   ...         NaN    NaN    NaN     NaN   
2         Philippines       5   ...         NaN    NaN    NaN     NaN   
3              Greece       8   ...         NaN    NaN    NaN     NaN   
4               Japan       4   ...         NaN    NaN    NaN     NaN   

   dbsource  INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0      PGIS        0         0        0       0      NaN  
1      PGIS        0         1        1       1      NaN  
2      PGIS       -9        -9        1       1      NaN  
3      PGIS       -9        -9        1       1      NaN  
4      PGIS       -9        -9        1       1      NaN  

[5 rows x 135 columns]

In [7]:
df.shape

(170350, 135)

In [8]:
df.columns

Index(['eventid', 'iyear', 'imonth', 'iday', 'approxdate', 'extended',
       'resolution', 'country', 'country_txt', 'region',
       ...
       'addnotes', 'scite1', 'scite2', 'scite3', 'dbsource', 'INT_LOG',
       'INT_IDEO', 'INT_MISC', 'INT_ANY', 'related'],
      dtype='object', length=135)

In [18]:
#restrict this dataset to occurrences in the US.
df1= df['country_txt'].str.contains('United States')
df2= df[df1]

In [19]:
#structure of DataFrame i need
df2.shape

(2758, 135)

In [31]:
df2.head()

eventid  iyear  imonth  iday approxdate  extended resolution  \
5   197001010002   1970       1     1        NaN         0        NaT   
7   197001020002   1970       1     2        NaN         0        NaT   
8   197001020003   1970       1     2        NaN         0        NaT   
9   197001030001   1970       1     3        NaN         0        NaT   
10  197001050001   1970       1     1        NaN         0        NaT   

    country    country_txt  region   ...     \
5       217  United States       1   ...      
7       217  United States       1   ...      
8       217  United States       1   ...      
9       217  United States       1   ...      
10      217  United States       1   ...      

                                             addnotes  \
5   The Cairo Chief of Police, William Petersen, r...   
7   Damages were estimated to be between $20,000-$...   
8   The New Years Gang issue a communiqué to a loc...   
9   Karl Armstrong's girlfriend, Lynn Schultz, dro...   
10                                                NaN   

                                               scite1  \
5   "Police Chief Quits," Washington Post, January...   
7   Committee on Government Operations United Stat...   
8   Tom Bates, "Rads: The 1970 Bombing of the Army...   
9   Committee on Government Operations United Stat...   
10                                                NaN   

                                               scite2  \
5   "Cairo Police Chief Quits; Decries Local 'Mili...   
7   Christopher Hewitt, "Political Violence and Te...   
8   David Newman, Sandra Sutherland, and Jon Stewa...   
9   Tom Bates, "Rads: The 1970 Bombing of the Army...   
10                                                NaN   

                                               scite3        dbsource  \
5   Christopher Hewitt, "Political Violence and Te...  Hewitt Project   
7                                                 NaN  Hewitt Project   
8   The Wisconsin Cartographers' Guild, "Wisconsin...  Hewitt Project   
9   David Newman, Sandra Sutherland, and Jon Stewa...  Hewitt Project   
10                                                NaN            PGIS   

    INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
5        -9        -9        0      -9      NaN  
7        -9        -9        0      -9      NaN  
8         0         0        0       0      NaN  
9         0         0        0       0      NaN  
10        0         0        0       0      NaN  

[5 rows x 135 columns]

In [33]:
#shows number missing values for each feature in the DataFrame
df2.isnull().sum()

eventid                  0
iyear                    0
imonth                   0
iday                     0
approxdate            2721
extended                 0
resolution            2746
country                  0
country_txt              0
region                   0
region_txt               0
provstate                0
city                     0
latitude                 1
longitude                1
specificity              0
vicinity                 0
location              1852
summary               1054
crit1                    0
crit2                    0
crit3                    0
doubtterr                0
alternative           2364
alternative_txt       2364
multiple                 0
success                  0
suicide                  0
attacktype1              0
attacktype1_txt          0
                      ... 
propextent            1191
propextent_txt        1191
propvalue             1840
propcomment           1748
ishostkid              176
nhostkid              2696
n

***Do i need all the 135 features? We'd see after running thorough EDA on this dataset. Taking a break!!!!***